<div style="background-color:#FFFFFF; padding:20px;">
  <h1 style="font-size:5em; font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif; font-weight:bold; color:#3E3E3E; text-align:center; text-shadow:2px 2px #F5F5F5; margin-bottom:0;">Marketing Campaign</h1>
  <p style="font-size:1.2em; font-family: 'Open Sans', Tahoma, Geneva, Verdana, sans-serif; font-weight:bold; color:#3E3E3E; text-align:center; margin-top:5px; margin-bottom:0;">Notebook Version: 1.0 | Author: Pranshu Chaurasia | Last Updated: April 15, 2023</p>
  <hr style="margin-top:5px; margin-bottom:5px;">
</div>

# Introduction
Customer Personality Analysis is a detailed analysis of a company’s ideal customers. It
helps a business to better understand its customers and makes it easier for them to
modify products according to the specific needs, behaviors and concerns of different
types of customers. </br>

# Problem Statement
Customer personality analysis helps a business to modify its product based on its target
customers from different types of customer segments. For example, instead of spending
money to market a new product to every customer in the company’s database, a
company can analyze which customer segment is most likely to buy the product and then
market the product only on that particular segment.

# Objective
The main objective is to train a predictive model which allows the company to maximize the profit of the next marketing campaign so that who will respond to an offer for a product or service.

# Importing the required Libraries for analysis

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

warnings.filterwarnings("ignore")

%matplotlib inline

# Imporing the dataset

In [5]:
data=pd.read_csv(r'marketing_campaign.csv')

In [7]:
# Creating a copy of pandas dataframe
df = data.copy()

In [8]:
#Displaying 10 ramdom rows from the data
df.sample(10)

ID  Year_Birth   Education Marital_Status   Income  Kidhome  \
1623   6097        1959  Graduation         Single  50664.0        1   
2095  11031        1950         PhD        Married  48699.0        1   
1338   2173        1952  Graduation          Widow  40049.0        0   
1534   3745        1988    2n Cycle        Married  23331.0        1   
1863   3254        1971  Graduation        Married  70886.0        1   
567     891        1951      Master       Together  29298.0        1   
1468   7196        1950         PhD        Married  41145.0        1   
1081    247        1971  Graduation        Married  36715.0        1   
715   10479        1975         PhD        Married  76618.0        0   
1734   2499        1958    2n Cycle         Single  75342.0        0   

      Teenhome Dt_Customer  Recency  MntWines  ...  NumWebVisitsMonth  \
1623         1  2013-01-16       64       313  ...                  9   
2095         1  2013-05-21       90        26  ...                  5   
1338         1  2013-10-16       23        13  ...                  6   
1534         0  2012-09-04       92         5  ...                  8   
1863         0  2013-10-14       65       407  ...                  2   
567          1  2013-09-07       60         6  ...                  5   
1468         1  2014-02-08       20         9  ...                  3   
1081         0  2012-11-17       16       172  ...                  9   
715          0  2012-12-07       64       749  ...                  1   
1734         1  2013-05-06       38       204  ...                  4   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
1623             0             0             0             0             0   
2095             0             0             0             0             0   
1338             0             0             0             0             0   
1534             0             0             0             0             0   
1863             0             0             0             0             0   
567              0             0             0             0             0   
1468             0             0             0             0             0   
1081             0             0             0             0             0   
715              0             0             0             0             0   
1734             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
1623         0              3         11         0  
2095         0              3         11         0  
1338         0              3         11         0  
1534         0              3         11         0  
1863         0              3         11         0  
567          0              3         11         0  
1468         0              3         11         0  
1081         0              3         11         1  
715          0              3         11         0  
1734         0              3         11         0  

[10 rows x 29 columns]

# Attribute Information:
<table>
<thead>
<tr>
<th>Column Name</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>AcceptedCmp1</strong></td>
<td>1 if customer accepted the offer in the 1st campaign, 0 otherwise</td>
</tr>
<tr>
<td><strong>AcceptedCmp2</strong></td>
<td>1 if customer accepted the offer in the 2nd campaign, 0 otherwise</td>
</tr>
<tr>
<td><strong>AcceptedCmp3</strong></td>
<td>1 if customer accepted the offer in the 3rd campaign, 0 otherwise</td>
</tr>
<tr>
<td><strong>AcceptedCmp4</strong></td>
<td>1 if customer accepted the offer in the 4th campaign, 0 otherwise</td>
</tr>
<tr>
<td><strong>AcceptedCmp5</strong></td>
<td>1 if customer accepted the offer in the 5th campaign, 0 otherwise</td>
</tr>
<tr>
<td><strong>Complain</strong></td>
<td>1 if customer complained in the last 2 years</td>
</tr>
<tr>
<td><strong>DtCustomer</strong></td>
<td>date of customer’s enrolment with the company</td>
</tr>
<tr>
<td><strong>Education</strong></td>
<td>customer’s level of education</td>
</tr>
<tr>
<td><strong>Marital_Status</strong></td>
<td>customer’s marital status</td>
</tr>
<tr>
<td><strong>Kidhome</strong></td>
<td>number of small children in customer’s household</td>
</tr>
<tr>
<td><strong>Teenhome</strong></td>
<td>number of teenagers in customer’s household</td>
</tr>
<tr>
<td><strong>Income</strong></td>
<td>customer’s yearly household income</td>
</tr>
<tr>
<td><strong>MntFishProducts</strong></td>
<td>amount spent on fish products in the last 2 years</td>
</tr>
<tr>
<td><strong>MntMeatProducts</strong></td>
<td>amount spent on meat products in the last 2 years</td>
</tr>
<tr>
<td><strong>MntFruits</strong></td>
<td>amount spent on fruits products in the last 2 years</td>
</tr>
<tr>
<td><strong>MntSweetProducts</strong></td>
<td>amount spent on sweet products in the last 2 years</td>
</tr>
<tr>
<td><strong>MntWines</strong></td>
<td>amount spent on wine products in the last 2 years</td>
</tr>
<tr>
<td><strong>MntGoldProds</strong></td>
<td>amount spent on gold products in the last 2 years</td>
</tr>
<tr>
<td><strong>NumDealsPurchases</strong></td>
<td>number of purchases made with discount</td>
</tr>
<tr>
<td><strong>NumCatalogPurchases</strong></td>
<td>number of purchases made using catalogue</td>
</tr>
<tr>
<td><strong>NumStorePurchases</strong></td>
<td>number of purchases made directly in stores</td>
</tr>
<tr>
<td><strong>NumWebPurchases</strong></td>
    <td>number of purchases made through company’s web site</td>
 <tr>
<tr>
<td><strong>Recency</strong></td>
<td>number of days since the last purchase</td>
</tr>

<tr>
<td><strong>NumWebVisitsMonth</strong></td>
<td>number of visits to company’s web site in the last month</td>
</tr>

<tr>
<td><strong>Year_Birth</strong></td>
<td>Birth Year of Customer</td>
</tr>

<tr>
<td><strong>ID</strong></td>
<td>Unique Id </td>
</tr>

<tr>
<td><strong>Z_CostContact</strong></td>
<td>Cost to contact a customer</td>
</tr>

<tr>
<td><strong>Z_Revenue</strong></td>
<td>Revenue after client accepting campaign</td>
</tr>

<td><strong>Response (Target Variable)</strong></td>
<td>1 if customer accepted the offer in the last campaign, 0 otherwise</td>
</tr>

# Exploratory Data Analysis

In [9]:
# Checking size of data
df.shape

(2240, 29)

We see that the total rows are 2240 and total columns are 29.

In [10]:
df.columns

Index(['ID', 'Year_Birth', 'Education', 'Marital_Status', 'Income', 'Kidhome',
       'Teenhome', 'Dt_Customer', 'Recency', 'MntWines', 'MntFruits',
       'MntMeatProducts', 'MntFishProducts', 'MntSweetProducts',
       'MntGoldProds', 'NumDealsPurchases', 'NumWebPurchases',
       'NumCatalogPurchases', 'NumStorePurchases', 'NumWebVisitsMonth',
       'AcceptedCmp3', 'AcceptedCmp4', 'AcceptedCmp5', 'AcceptedCmp1',
       'AcceptedCmp2', 'Complain', 'Z_CostContact', 'Z_Revenue', 'Response'],
      dtype='object')

In [11]:
df['Dt_Customer']

0       2012-09-04
1       2014-03-08
2       2013-08-21
3       2014-02-10
4       2014-01-19
           ...    
2235    2013-06-13
2236    2014-06-10
2237    2014-01-25
2238    2014-01-24
2239    2012-10-15
Name: Dt_Customer, Length: 2240, dtype: object